tools/apps/src/bin/quiche-client.rs

```diff
@@ -0,0 +1,790 @@
+                            make_fd(&req.url, &path, req.cardinal),
+                        );
+                    }
+
+                    *stream_id += 4;
+
+                    reqs_done += 1;
+                }
+
+                *reqs_sent += reqs_done;
+            },
+
+            HttpConn::Http3Conn {
+                h3_conn,
+                reqs_sent,
+                reqs,
+                body,
+                fds,
+                ..
+            } => {
+                let mut reqs_done = 0;
+
+                for req in reqs.iter().skip(*reqs_sent) {
+                    info!("sending HTTP request {:?}", req.hdrs);
+
+                    let s = match h3_conn.send_request(
+                        conn,
+                        &req.hdrs,
+                        body.is_none(),
+                    ) {
+                        Ok(v) => v,
+
+                        Err(quiche::h3::Error::TransportError(
+                            quiche::Error::StreamLimit,
+                        )) => {
+                            debug!("not enough stream credits, retry later...");
+                            break;
+                        },
+
+                        Err(e) => {
+                            error!("failed to send request {:?}", e);
+                            break;
+                        },
+                    };
+
+                    if let Some(path) = target_path.as_ref() {
+                        fds.insert(s, make_fd(&req.url, path, req.cardinal));
+                    }
+
+                    if let Some(body) = &body {
+                        if let Err(e) = h3_conn.send_body(conn, s, body, true) {
+                            error!("failed to send request body {:?}", e);
+                            break;
+                        }
+                    }
+
+                    reqs_done += 1;
+                }
+
+                *reqs_sent += reqs_done;
+            },
+        }
+    }
+
+    fn handle_responses(
+        &mut self, conn: &mut quiche::Connection, buf: &mut [u8],
+        req_start: &std::time::Instant,
+    ) {
+        match self {
+            HttpConn::Http09Conn {
+                fds,
+                reqs_complete,
+                reqs,
+                ..
+            } => {
+                // Process all readable streams.
+                for s in conn.readable() {
+                    while let Ok((read, fin)) = conn.stream_recv(s, buf) {
+                        debug!("received {} bytes", read);
+
+                        let stream_buf = &buf[..read];
+
+                        debug!(
+                            "stream {} has {} bytes (fin? {})",
+                            s,
+                            stream_buf.len(),
+                            fin
+                        );
+
+                        print!("{}", unsafe {
+                            std::str::from_utf8_unchecked(&stream_buf)
+                        });
```

Comment:
Since we have `--dump-responses` option, I think when we set this option, no need to print the body. to stdout

Reply:
it matches the behaviour of the example apps and is useful to see stuff happen when running interactively. Happy to make this more conditional if we really want

Reply:
When quiche-clients runs with multiple urls, stdout output can be mixed up -- so need to use --dump-responses option any way to verify. Also in interop runner, log file will contain all body dump, make it harder to look at -- found during I run interop script using this branch.

Reply:
Agreed. I'll make the change to not print if `--dump-response` is set.